### PR TITLE
publishing to mhub on tournament setting change

### PR DESCRIPTION
**Pull request by taltaub22**
- refactoring of mhub connection
- refactoring of general setting router
- publishing to mhub on tournament setting change

### CODE REVIEW

File: Utils/mhubConnection.js [hunk 1]
```diff
@@ -12,22 +12,48 @@ const MHUB_NODES = {
...
 
 const mhubClient = new MClient(process.env.MHUB_URI)
-mhubClient.login('protected-client', process.env.PROTECTED_MHUB_PASSWORD)
 
 mhubClient.on('error', msg => {
   MsLogger.error('Unable to connect to mhub, other modules won\'t be notified changes \n ' + msg)
 })
 
-function publishUpdateMsg(nameSpace) {
+function publishUpdateMsg (nameSpace) {
```
Comment (idanstark42) [inline Utils/mhubConnection.js]: This function shuld use `publishMsg` to avoid code repetition

File: logic/tournamentSettingsLogic.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+function updateSetting (settingName, value) {
+  return MongoClient.connect(MONGU_URI).then(connection => {
+    const setDocument = {}
+    setDocument[settingName] = value
+    return connection.db().collection(SETTING_COLLECTION_NAME)
+      .findOneAndUpdate({}, {$set: setDocument})
+      .then(dbResponse => {
+        if (dbResponse.ok == 1) {
+
+          if(settingName == 'tournamentLevel'){
```
Comment (idanstark42) [inline logic/tournamentSettingsLogic.js]: You should publish the title as well using `title:updated` and givving its value

Comment (taltaub22) [reply logic/tournamentSettingsLogic.js]: could it be the full setting name?

Comment (idanstark42) [reply logic/tournamentSettingsLogic.js]: Sure, but change the `tournamentLevel` to `tournamentStage` because it's called stage everywhere else

Comment (taltaub22) [reply logic/tournamentSettingsLogic.js]: Yes, I Agree on that

**Comment by taltaub22** [comment]
I'm done with the fixes @idanstark42

**Pull request merged**